dallinger/experiment_server/experiment_server.py

```diff
@@ -532,10 +546,10 @@ def prepare_advertisement():
         if already_participated:
             raise ExperimentError("already_did_exp_hit")
 
-    recruiter_name = request.args.get("recruiter")
-    if not recruiter_name:
-        recruiter = recruiters.from_config(config)
-        recruiter_name = recruiter.nickname
+    if recruiter_name == "prolific":
```

Comment:
I think these lines should be moved up to around line 533?

Reply:
Yep, moved them up.